dm/relay/relay.go

```diff
@@ -558,6 +559,17 @@ func (r *Relay) handleEvents(
 			}
 		}
 
+		if firstEvent {
```

Comment:
seems don't need `firstEvent` var, we can  move this logic to L549 it already check the event is `fakerotateEvent`  with `if _, ok := e.Event.(*replication.RotateEvent); ok && utils.IsFakeRotateEvent(e.Header)`

Reply:
fake rorate events are sent in many cases, not just on connect.

Reply:
https://github.com/mysql/mysql-server/blob/3290a66c89eb1625a7058e0ef732432b6952b435/sql/rpl_binlog_sender.cc#L395-L402

seems there's no other cases of sending fake rotate events, so i also agree we move this code inside above `IsFakeRotateEvent` check

Reply:
see https://github.com/mysql/mysql-server/blob/3290a66c89eb1625a7058e0ef732432b6952b435/sql/rpl_binlog_sender.h#L247-L273